toolchain/sem_ir/inst_kind.def

```diff
@@ -23,6 +23,7 @@ CARBON_SEM_IR_INST_KIND(ArrayIndex)
 CARBON_SEM_IR_INST_KIND(ArrayInit)
 CARBON_SEM_IR_INST_KIND(ArrayType)
 CARBON_SEM_IR_INST_KIND(Assign)
+CARBON_SEM_IR_INST_KIND(AssociatedConstantDecl)
```

Comment:
As you're adding this, would it be worth adding a todo test for import? i.e., in file A, declare an interface with an associated constant; then in file B, reference the interface (e.g., with an alias). The result should be that the interface members, including the associated constant, try to resolve; I'd expect that to currently generate a TODO (from import_ref, without code changes).

Reply:
Done.